src/dune_rules/pkg_rules.ml

```diff
@@ -514,6 +516,80 @@ module Substitute = struct
 end
 
 module Run_with_path = struct
+  module Output : sig
+    type error
+
+    val io : error -> Process.Io.output Process.Io.t
+
+    val with_error
+      :  accepted_exit_codes:int Predicate.t
+      -> pkg_name:Dune_pkg.Package_name.t option
+      -> display:Display.t
+      -> (error -> 'a)
+      -> 'a
+
+    val prerr : loc:Loc.t -> rc:int -> error -> unit
+  end = struct
+    type error =
+      { pkg_name : Dune_pkg.Package_name.t option
+      ; filename : Dpath.t
+      ; io : Process.Io.output Process.Io.t
+      ; accepted_exit_codes : int Predicate.t
+      ; display : Display.t
+      }
+
+    let io t = t.io
+
+    let with_error ~accepted_exit_codes ~pkg_name ~display f =
+      let filename = Temp.create File ~prefix:"dune-pkg" ~suffix:"stderr" in
```

Comment:
This is okay in this PR, yet I think it could be useful to write these files into `_build` instead, similarly how we write `_build/log` into the build directory and not to a temporary file. That way users have the option to navigate to the folder containing the log output and see what the outputs were.

Similarly, it would also be interesting to have the `stdout` captured in the same way, for example output of `./configure` style scripts (to answer "was the library successfully detected etc.)

Reply:
I agree with your proposal, as it would be nice to be able to get the error logs after the `build`. However, it might be something we want to do in another PR as the scope of this one would be too wide.